### PR TITLE
feat: respect the given executor

### PR DESCRIPTION
**Pull request by KazuCocoa**
Got a request to use the given executor if the first argument was the instance as same as Selenium does.

Example usage is in the README.

### CODE REVIEW

File: appium/webdriver/webdriver.py [hunk 1]
```diff
@@ -228,8 +228,11 @@ def __init__(
             AppiumConnection.set_certificate_bundle_path(None)
             urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
+        if isinstance(command_executor, (str, bytes)):
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py:231]: can it be `bytes`?

File: appium/webdriver/appium_connection.py [hunk 2]
```diff
@@ -25,21 +25,38 @@
...
 class AppiumConnection(RemoteConnection):
-    def _get_connection_manager(self) -> Union[urllib3.PoolManager, urllib3.ProxyManager]:
-        # https://github.com/SeleniumHQ/selenium/blob/0e0194b0e52a34e7df4b841f1ed74506beea5c3e/py/selenium/webdriver/remote/remote_connection.py#L134
-        pool_manager_init_args = {'timeout': self._timeout}
+    def __init__(
+        self, remote_server_addr: Union[str, bytes], keep_alive: bool = False, ignore_proxy: Optional[bool] = False
+    ):
+        self._pool_manager_init_args: Dict[str, Any] = {}
+
+        super().__init__(remote_server_addr, keep_alive=keep_alive, ignore_proxy=ignore_proxy)
```
Comment (mykola-mokhnach) [inline appium/webdriver/appium_connection.py:33]: I would rather call super fist

File: appium/webdriver/appium_connection.py [hunk 3]
```diff
@@ -25,21 +25,38 @@
 
 
 class AppiumConnection(RemoteConnection):
-    def _get_connection_manager(self) -> Union[urllib3.PoolManager, urllib3.ProxyManager]:
-        # https://github.com/SeleniumHQ/selenium/blob/0e0194b0e52a34e7df4b841f1ed74506beea5c3e/py/selenium/webdriver/remote/remote_connection.py#L134
-        pool_manager_init_args = {'timeout': self._timeout}
+    def __init__(
+        self, remote_server_addr: Union[str, bytes], keep_alive: bool = False, ignore_proxy: Optional[bool] = False
+    ):
+        self._pool_manager_init_args: Dict[str, Any] = {}
```
Comment (mykola-mokhnach) [inline appium/webdriver/appium_connection.py:31]: not sure if we need to set the default value here if we anyway redefine it in set_init_args_for_pool_manager

File: appium/webdriver/webdriver.py (hunk 1 above)
Comment (KazuCocoa) [reply appium/webdriver/webdriver.py:231]: Added bytes as selenium had, but hm, maybe my reference was old one. Latest selenium is str

File: test/unit/webdriver/webdriver_test.py [hunk 4]
```diff
@@ -298,6 +300,95 @@ def add_command(self):
...
+        }
+
+        class CustomAppiumConnection(AppiumConnection):
+            # To explicitly check if the given executor is used
+            pass
+
+        init_args_for_pool_manager = {'retries': urllib3.util.retry.Retry(total=3, connect=3, read=False)}
+        custom_appium_connection = CustomAppiumConnection(
+            # keep alive has different route to set init args for the pool manager
+            keep_alive=True,
```
Comment (KazuCocoa) [inline test/unit/webdriver/webdriver_test.py:366]: Noticed the old method did not work when keep_alive was enabled, so modified to pass the arguments as the init

File: README.md [hunk 5]
```diff
@@ -192,10 +192,63 @@ options.automation_name = 'safari'
...
+from appium.webdriver.appium_connection import AppiumConnection
+
+# Retry connection error up to 3 times.
+init_args_for_pool_manage = {
+    'retries': urllib3.util.retry.Retry(total=3, connect=3, read=False)
+}
+appium_executor = AppiumConnection(remote_server_addr='http://127.0.0.1:4723',
+    init_args_for_pool_manage=init_args_for_pool_manage)
+
+options = XCUITestOptions().load_capabilities({
```
Comment (mykola-mokhnach) [inline README.md:219]: load_capabilities was created rather for compatibility with older stuff. Consider using appropriate setters and getters instead

File: appium/webdriver/appium_connection.py [hunk 6]
```diff
@@ -25,17 +25,35 @@
...
+    def __init__(
+        self,
+        remote_server_addr: str,
+        keep_alive: bool = False,
+        ignore_proxy: Optional[bool] = False,
+        init_args_for_pool_manager: Union[Dict[str, Any], None] = None,
+    ):
+
+        # Need to call before super().__init__ in order to pass arguments for the pool manager in the super.
+        if init_args_for_pool_manager is None:
```
Comment (mykola-mokhnach) [inline appium/webdriver/appium_connection.py:37]: could be simplified to `self._init_args_for_pool_manager = init_args_for_pool_manager or {}`

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**